{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os \n",
    "import subprocess\n",
    "import pandas as pd "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "def retrive_ref(tax,taxid=False):\n",
    "    if taxid == False:\n",
    "        out = subprocess.Popen(f\"esearch -db assembly -query '{tax}[ORGN]' | esummary | xtract -pattern DocumentSummary \\\n",
    "                -def NA -element Organism,AssemblyAccession,Taxid,assembly-status -block Stat \\\n",
    "                -if Stat@category -equals contig_count -or Stat@category -equals contig_l50 \\\n",
    "                -or Stat@category -equals contig_n50 -or Stat@category -equals total_length \\\n",
    "                -element Stat\", shell=True, stdout=subprocess.PIPE).communicate()[0].decode('utf-8')\n",
    "    else:\n",
    "        tax=int(tax)\n",
    "        out = subprocess.Popen(f\"esearch -db assembly -query 'taxid{tax}' | esummary | xtract -pattern DocumentSummary \\\n",
    "                -def NA -element Organism,AssemblyAccession,Taxid,assembly-status -block Stat \\\n",
    "                -if Stat@category -equals contig_count -or Stat@category -equals contig_l50 \\\n",
    "                -or Stat@category -equals contig_n50 -or Stat@category -equals total_length \\\n",
    "                -element Stat\", shell=True, stdout=subprocess.PIPE).communicate()[0].decode('utf-8')\n",
    "    print(out)\n",
    "    raw_lst = []\n",
    "    lst = out.strip().split(\"\\n\")\n",
    "    for i in lst:\n",
    "        raw_lst.append(i.split(\"\\t\"))\n",
    "    df = pd.DataFrame(raw_lst)\n",
    "    df.columns =['Organism', 'AssemblyAccession', \n",
    "                    'Taxid', 'assembly-status', 'contig_count',\n",
    "                    'contig_l50', 'contig_n50', 'total_length']\n",
    "\n",
    "    df[\"Organism\"].unique()\n",
    "    df = df.astype({\"contig_count\": int})\n",
    "    return df\n",
    "\n",
    "def filter_taxa_df(df):\n",
    "    print(\"Please Select a Assembly level: \")\n",
    "    counter = 1\n",
    "    uniques = df[\"assembly-status\"].unique()\n",
    "    for i in uniques:\n",
    "        print(f\"{counter}: {i}\")\n",
    "        counter += 1\n",
    "    n = int(input(\"\\n\"))\n",
    "    m = n-1\n",
    "\n",
    "    out = df[df[\"assembly-status\"] == df[\"assembly-status\"].unique()[m]].sort_values(\"contig_count\")\n",
    "    \n",
    "    print(\"Please Select a Number for arganism: \")\n",
    "    counter = 1\n",
    "    uniques = out[\"Organism\"].unique()\n",
    "    for i in uniques:\n",
    "        print(f\"{counter}: {i}\")\n",
    "        counter += 1\n",
    "    n = int(input(\"\\n\"))\n",
    "    m = n-1\n",
    "    out2 = out[out[\"Organism\"] == out[\"Organism\"].unique()[m]].sort_values(\"contig_count\")\n",
    "    return out2\n",
    "\n",
    "def download_ref(df):\n",
    "    df2 = df[[\"AssemblyAccession\", \"total_length\", \"assembly-status\", \"contig_count\", \"contig_l50\", \"contig_n50\"]]\n",
    "    df2.index = df2.reset_index(drop = True).index + 1\n",
    "    print(\"Please Select a Number to Download ref: \")\n",
    "    print(df2)\n",
    "    n = int(input(\"\\n\"))\n",
    "    acc = test2.iloc[n][\"AssemblyAccession\"]\n",
    "    subprocess.Popen(f\"esearch -db assembly -query {acc} | elink -target nucleotide -name assembly_nuccore_refseq | efetch -format fasta > {acc}.fa\", shell=True,stdout=subprocess.PIPE)\n",
    "    print(\"Done\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "ename": "ValueError",
     "evalue": "Length mismatch: Expected axis has 1 elements, new values have 8 elements",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-12-7a81dfdb5baa>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0;31m# df = retrive_ref(\"Agrobacterium tumefaciens\")\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m \u001b[0mretrive_ref\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;36m358\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mtaxid\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m<ipython-input-11-1d028cb4cb93>\u001b[0m in \u001b[0;36mretrive_ref\u001b[0;34m(tax, taxid)\u001b[0m\n\u001b[1;32m     19\u001b[0m         \u001b[0mraw_lst\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msplit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"\\t\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     20\u001b[0m     \u001b[0mdf\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mDataFrame\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mraw_lst\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 21\u001b[0;31m     df.columns =['Organism', 'AssemblyAccession', \n\u001b[0m\u001b[1;32m     22\u001b[0m                     \u001b[0;34m'Taxid'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'assembly-status'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'contig_count'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     23\u001b[0m                     'contig_l50', 'contig_n50', 'total_length']\n",
      "\u001b[0;32m~/.local/lib/python3.8/site-packages/pandas/core/generic.py\u001b[0m in \u001b[0;36m__setattr__\u001b[0;34m(self, name, value)\u001b[0m\n\u001b[1;32m   5489\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   5490\u001b[0m             \u001b[0mobject\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__getattribute__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mname\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 5491\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mobject\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__setattr__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mname\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   5492\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0mAttributeError\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   5493\u001b[0m             \u001b[0;32mpass\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.local/lib/python3.8/site-packages/pandas/_libs/properties.pyx\u001b[0m in \u001b[0;36mpandas._libs.properties.AxisProperty.__set__\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32m~/.local/lib/python3.8/site-packages/pandas/core/generic.py\u001b[0m in \u001b[0;36m_set_axis\u001b[0;34m(self, axis, labels)\u001b[0m\n\u001b[1;32m    761\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_set_axis\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mint\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlabels\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mIndex\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m->\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    762\u001b[0m         \u001b[0mlabels\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mensure_index\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlabels\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 763\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_mgr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mset_axis\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0maxis\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlabels\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    764\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_clear_item_cache\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    765\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.local/lib/python3.8/site-packages/pandas/core/internals/managers.py\u001b[0m in \u001b[0;36mset_axis\u001b[0;34m(self, axis, new_labels)\u001b[0m\n\u001b[1;32m    214\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mset_axis\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mint\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnew_labels\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mIndex\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m->\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    215\u001b[0m         \u001b[0;31m# Caller is responsible for ensuring we have an Index object.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 216\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_validate_set_axis\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0maxis\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnew_labels\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    217\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0maxes\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0maxis\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mnew_labels\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    218\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.local/lib/python3.8/site-packages/pandas/core/internals/base.py\u001b[0m in \u001b[0;36m_validate_set_axis\u001b[0;34m(self, axis, new_labels)\u001b[0m\n\u001b[1;32m     55\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     56\u001b[0m         \u001b[0;32melif\u001b[0m \u001b[0mnew_len\u001b[0m \u001b[0;34m!=\u001b[0m \u001b[0mold_len\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 57\u001b[0;31m             raise ValueError(\n\u001b[0m\u001b[1;32m     58\u001b[0m                 \u001b[0;34mf\"Length mismatch: Expected axis has {old_len} elements, new \"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     59\u001b[0m                 \u001b[0;34mf\"values have {new_len} elements\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mValueError\u001b[0m: Length mismatch: Expected axis has 1 elements, new values have 8 elements"
     ]
    }
   ],
   "source": [
    "# df = retrive_ref(\"Agrobacterium tumefaciens\")\n",
    "retrive_ref(358,taxid=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Please Select a Assembly level: \n",
      "1: Contig\n",
      "2: Chromosome\n",
      "3: Complete Genome\n",
      "4: Scaffold\n",
      "Please Select a Number for arganism: \n",
      "1: Agrobacterium tumefaciens (a-proteobacteria)\n",
      "2: Agrobacterium radiobacter K84 (a-proteobacteria)\n"
     ]
    }
   ],
   "source": [
    "test = filter_taxa_df(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Organism</th>\n",
       "      <th>AssemblyAccession</th>\n",
       "      <th>Taxid</th>\n",
       "      <th>assembly-status</th>\n",
       "      <th>contig_count</th>\n",
       "      <th>contig_l50</th>\n",
       "      <th>contig_n50</th>\n",
       "      <th>total_length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Agrobacterium tumefaciens (a-proteobacteria)</td>\n",
       "      <td>GCF_017726655.1</td>\n",
       "      <td>358</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2913790</td>\n",
       "      <td>5082709</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Agrobacterium tumefaciens (a-proteobacteria)</td>\n",
       "      <td>GCF_014489975.1</td>\n",
       "      <td>358</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2922353</td>\n",
       "      <td>5109662</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>205</th>\n",
       "      <td>Agrobacterium tumefaciens (a-proteobacteria)</td>\n",
       "      <td>GCF_001551895.1</td>\n",
       "      <td>358</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2983661</td>\n",
       "      <td>5481595</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Agrobacterium tumefaciens (a-proteobacteria)</td>\n",
       "      <td>GCA_017897985.1</td>\n",
       "      <td>358</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>2773415</td>\n",
       "      <td>5529348</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>168</th>\n",
       "      <td>Agrobacterium tumefaciens (a-proteobacteria)</td>\n",
       "      <td>GCF_005221425.1</td>\n",
       "      <td>358</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>2953857</td>\n",
       "      <td>5526017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>177</th>\n",
       "      <td>Agrobacterium tumefaciens (a-proteobacteria)</td>\n",
       "      <td>GCF_003667905.1</td>\n",
       "      <td>358</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>3027766</td>\n",
       "      <td>5446032</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>176</th>\n",
       "      <td>Agrobacterium tumefaciens (a-proteobacteria)</td>\n",
       "      <td>GCF_003667925.1</td>\n",
       "      <td>358</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>2096799</td>\n",
       "      <td>5939134</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>178</th>\n",
       "      <td>Agrobacterium tumefaciens (a-proteobacteria)</td>\n",
       "      <td>GCF_003666465.1</td>\n",
       "      <td>358</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>2095493</td>\n",
       "      <td>5870997</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170</th>\n",
       "      <td>Agrobacterium tumefaciens (a-proteobacteria)</td>\n",
       "      <td>GCF_005221385.1</td>\n",
       "      <td>358</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>3053646</td>\n",
       "      <td>5515917</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>179</th>\n",
       "      <td>Agrobacterium tumefaciens (a-proteobacteria)</td>\n",
       "      <td>GCF_003666445.1</td>\n",
       "      <td>358</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>2867449</td>\n",
       "      <td>5683743</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152</th>\n",
       "      <td>Agrobacterium tumefaciens (a-proteobacteria)</td>\n",
       "      <td>GCF_002591665.3</td>\n",
       "      <td>358</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>2943916</td>\n",
       "      <td>5720291</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>Agrobacterium tumefaciens (a-proteobacteria)</td>\n",
       "      <td>GCF_013401495.1</td>\n",
       "      <td>358</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>2861634</td>\n",
       "      <td>5677928</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Agrobacterium tumefaciens (a-proteobacteria)</td>\n",
       "      <td>GCF_016403145.1</td>\n",
       "      <td>358</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>2088748</td>\n",
       "      <td>5932670</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>161</th>\n",
       "      <td>Agrobacterium tumefaciens (a-proteobacteria)</td>\n",
       "      <td>GCF_002591665.2</td>\n",
       "      <td>358</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>2943916</td>\n",
       "      <td>5725540</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>184</th>\n",
       "      <td>Agrobacterium tumefaciens (a-proteobacteria)</td>\n",
       "      <td>GCF_002943835.1</td>\n",
       "      <td>358</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>3058772</td>\n",
       "      <td>5985137</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>180</th>\n",
       "      <td>Agrobacterium tumefaciens (a-proteobacteria)</td>\n",
       "      <td>GCF_003666425.1</td>\n",
       "      <td>358</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>2106029</td>\n",
       "      <td>5767533</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>169</th>\n",
       "      <td>Agrobacterium tumefaciens (a-proteobacteria)</td>\n",
       "      <td>GCF_005221405.1</td>\n",
       "      <td>358</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "      <td>2421208</td>\n",
       "      <td>6469646</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>171</th>\n",
       "      <td>Agrobacterium tumefaciens (a-proteobacteria)</td>\n",
       "      <td>GCF_005221345.1</td>\n",
       "      <td>358</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "      <td>2262814</td>\n",
       "      <td>5996243</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>167</th>\n",
       "      <td>Agrobacterium tumefaciens (a-proteobacteria)</td>\n",
       "      <td>GCF_005221445.1</td>\n",
       "      <td>358</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>2911401</td>\n",
       "      <td>5525862</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>166</th>\n",
       "      <td>Agrobacterium tumefaciens (a-proteobacteria)</td>\n",
       "      <td>GCF_005221465.1</td>\n",
       "      <td>358</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>3053995</td>\n",
       "      <td>5839776</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>172</th>\n",
       "      <td>Agrobacterium tumefaciens (a-proteobacteria)</td>\n",
       "      <td>GCF_005221325.1</td>\n",
       "      <td>358</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "      <td>2301325</td>\n",
       "      <td>6138594</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Agrobacterium tumefaciens (a-proteobacteria)</td>\n",
       "      <td>GCA_017584185.1</td>\n",
       "      <td>358</td>\n",
       "      <td>Complete Genome</td>\n",
       "      <td>7</td>\n",
       "      <td>2</td>\n",
       "      <td>2146132</td>\n",
       "      <td>6305579</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                         Organism AssemblyAccession Taxid  \\\n",
       "15   Agrobacterium tumefaciens (a-proteobacteria)   GCF_017726655.1   358   \n",
       "18   Agrobacterium tumefaciens (a-proteobacteria)   GCF_014489975.1   358   \n",
       "205  Agrobacterium tumefaciens (a-proteobacteria)   GCF_001551895.1   358   \n",
       "5    Agrobacterium tumefaciens (a-proteobacteria)   GCA_017897985.1   358   \n",
       "168  Agrobacterium tumefaciens (a-proteobacteria)   GCF_005221425.1   358   \n",
       "177  Agrobacterium tumefaciens (a-proteobacteria)   GCF_003667905.1   358   \n",
       "176  Agrobacterium tumefaciens (a-proteobacteria)   GCF_003667925.1   358   \n",
       "178  Agrobacterium tumefaciens (a-proteobacteria)   GCF_003666465.1   358   \n",
       "170  Agrobacterium tumefaciens (a-proteobacteria)   GCF_005221385.1   358   \n",
       "179  Agrobacterium tumefaciens (a-proteobacteria)   GCF_003666445.1   358   \n",
       "152  Agrobacterium tumefaciens (a-proteobacteria)   GCF_002591665.3   358   \n",
       "36   Agrobacterium tumefaciens (a-proteobacteria)   GCF_013401495.1   358   \n",
       "17   Agrobacterium tumefaciens (a-proteobacteria)   GCF_016403145.1   358   \n",
       "161  Agrobacterium tumefaciens (a-proteobacteria)   GCF_002591665.2   358   \n",
       "184  Agrobacterium tumefaciens (a-proteobacteria)   GCF_002943835.1   358   \n",
       "180  Agrobacterium tumefaciens (a-proteobacteria)   GCF_003666425.1   358   \n",
       "169  Agrobacterium tumefaciens (a-proteobacteria)   GCF_005221405.1   358   \n",
       "171  Agrobacterium tumefaciens (a-proteobacteria)   GCF_005221345.1   358   \n",
       "167  Agrobacterium tumefaciens (a-proteobacteria)   GCF_005221445.1   358   \n",
       "166  Agrobacterium tumefaciens (a-proteobacteria)   GCF_005221465.1   358   \n",
       "172  Agrobacterium tumefaciens (a-proteobacteria)   GCF_005221325.1   358   \n",
       "16   Agrobacterium tumefaciens (a-proteobacteria)   GCA_017584185.1   358   \n",
       "\n",
       "     assembly-status  contig_count contig_l50 contig_n50 total_length  \n",
       "15   Complete Genome             2          1    2913790      5082709  \n",
       "18   Complete Genome             2          1    2922353      5109662  \n",
       "205  Complete Genome             2          1    2983661      5481595  \n",
       "5    Complete Genome             3          1    2773415      5529348  \n",
       "168  Complete Genome             3          1    2953857      5526017  \n",
       "177  Complete Genome             3          1    3027766      5446032  \n",
       "176  Complete Genome             4          2    2096799      5939134  \n",
       "178  Complete Genome             4          2    2095493      5870997  \n",
       "170  Complete Genome             4          1    3053646      5515917  \n",
       "179  Complete Genome             4          1    2867449      5683743  \n",
       "152  Complete Genome             4          1    2943916      5720291  \n",
       "36   Complete Genome             4          1    2861634      5677928  \n",
       "17   Complete Genome             4          2    2088748      5932670  \n",
       "161  Complete Genome             4          1    2943916      5725540  \n",
       "184  Complete Genome             5          1    3058772      5985137  \n",
       "180  Complete Genome             5          2    2106029      5767533  \n",
       "169  Complete Genome             6          2    2421208      6469646  \n",
       "171  Complete Genome             6          2    2262814      5996243  \n",
       "167  Complete Genome             6          1    2911401      5525862  \n",
       "166  Complete Genome             6          1    3053995      5839776  \n",
       "172  Complete Genome             6          2    2301325      6138594  \n",
       "16   Complete Genome             7          2    2146132      6305579  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "test2 = test[[\"AssemblyAccession\", \"total_length\", \"assembly-status\", \"contig_count\", \"contig_l50\", \"contig_n50\", \"FtpPath\"]]\n",
    "test2.index = test2.reset_index(drop = True).index + 1\n",
    "test2\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "acc = test2.iloc[0][\"AssemblyAccession\"]\n",
    "acc\n",
    "# subprocess.Popen(f\"esearch -db assembly -query {acc} | elink -target nucleotide -name assembly_nuccore_refseq | efetch -format fasta > {acc}.fa\", shell=True,stdout=subprocess.PIPE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "name = \"ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/286/275/GCA_000286275.1_ASM28627v1\"\n",
    "name = name + \"/\" + name.split(\"/\")[-1] + \"_genomic.fna.gz/\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/286/275/GCA_000286275.1_ASM28627v1/GCA_000286275.1_ASM28627v1_genomic.fna.gz/'"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "name"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "genomes/all/GCA/000/286/275/GCA_000286275.1_ASM28627v1/GCA_000286275.1_ASM28627v1_genomic.fna.gz"
   ]
  }
 ],
 "metadata": {
  "interpreter": {
   "hash": "e3381db70853396d7451af2db8736d8d09d94e70e95c0959b02ce6e6657e7302"
  },
  "kernelspec": {
   "display_name": "Python 3.8.5 ('base')",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
